src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Autodesk.DesignScript.Runtime;
+using Dynamo.Utilities;
+using Python.Runtime;
+
+namespace DSCPython
+{
+    [SupressImportIntoVM]
+    public enum EvaluationState { Begin, Success, Failed }
+
+    [SupressImportIntoVM]
+    public delegate void EvaluationEventHandler(EvaluationState state,
+                                                PyScope scope,
+                                                string code,
+                                                IList bindingValues);
+
+    /// <summary>
+    ///     Evaluates a Python script in the Dynamo context.
+    /// </summary>
+    [IsVisibleInDynamoLibrary(false)]
+    public static class CPythonEvaluator
+    {
+        /// <summary> stores a copy of the previously executed code</summary>
+        private static string prev_code { get; set; }
+
+        /// <summary>
+        ///     Executes a Python script with custom variable names. Script may be a string
+        ///     read from a file, for example. Pass a list of names (matching the variable
+        ///     names in the script) to bindingNames and pass a corresponding list of values
+        ///     to bindingValues.
+        /// </summary>
+        /// <param name="code">Python script as a string.</param>
+        /// <param name="bindingNames">Names of values referenced in Python script.</param>
+        /// <param name="bindingValues">Values referenced in Python script.</param>
+        public static object EvaluatePythonScript(
+            string code,
+            IList bindingNames,
+            [ArbitraryDimensionArrayImport] IList bindingValues)
+        {
+            // TODO - it would be nice if users could modify a preference
+            // setting enabling the ability to load additional paths
+
+            // Container for paths that will be imported in the PythonEngine
+            //List<string> paths = new List<string>();
+
+            // Attempt to get the Standard Python Library
+            //string stdLib = pythonStandardLibPath();
```

Comment:
Do we still need any of these?

Reply:
Same question

Reply:
I will address this in the follow-up PR.

Reply:
Thank you @reddyashish